spec/models/recipe_ingredients_spec.rb

```diff
@@ -6,4 +6,56 @@
+
+      @ingredient_4 = Ingredient.create!(name: "Sauce", cost: 4)
+      @ingredient_5 = Ingredient.create!(name: "Spaghetti", cost: 2)
+      @ingredient_6 = Ingredient.create!(name: "Ground Beef", cost: 7)
+
+      @ingredient_7 = Ingredient.create!(name: "Salt", cost: 4)
+      @ingredient_8 = Ingredient.create!(name: "Pepper", cost: 4)
+      @ingredient_9 = Ingredient.create!(name: "Avocado Oil", cost: 4)
+
+      @recipe_1 = Recipe.create!(name: "Tangy Salmon", complexity: 5, genre: "Mediterranean")
+      @recipe_2 = Recipe.create!(name: "Tried and True", complexity: 2, genre: "Italian")
+      @recipe_3 = Recipe.create!(name: "Avo Asparagus", complexity: 1, genre: "Sides")
+
+      # Tangy Salmon
+      @rec_ing_1 = RecipeIngredient.create!(recipe_id: @recipe_1.id, ingredient_id: @ingredient_1.id)
+      @rec_ing_2 = RecipeIngredient.create!(recipe_id: @recipe_1.id, ingredient_id: @ingredient_2.id)
+      @rec_ing_3 = RecipeIngredient.create!(recipe_id: @recipe_1.id, ingredient_id: @ingredient_3.id)
+      @rec_ing_4 = RecipeIngredient.create!(recipe_id: @recipe_1.id, ingredient_id: @ingredient_7.id)
+      @rec_ing_5 = RecipeIngredient.create!(recipe_id: @recipe_1.id, ingredient_id: @ingredient_8.id)
+      @rec_ing_6 = RecipeIngredient.create!(recipe_id: @recipe_1.id, ingredient_id: @ingredient_9.id)
+
+      # Tried and True
+      @rec_ing_7 = RecipeIngredient.create!(recipe_id: @recipe_2.id, ingredient_id: @ingredient_4.id)
+      @rec_ing_8 = RecipeIngredient.create!(recipe_id: @recipe_2.id, ingredient_id: @ingredient_5.id)
+      @rec_ing_9 = RecipeIngredient.create!(recipe_id: @recipe_2.id, ingredient_id: @ingredient_6.id)
+      @rec_ing_10 = RecipeIngredient.create!(recipe_id: @recipe_2.id, ingredient_id: @ingredient_7.id)
+      @rec_ing_11 = RecipeIngredient.create!(recipe_id: @recipe_2.id, ingredient_id: @ingredient_8.id)
+
+      # Avo Asparagus
+      @rec_ing_12 = RecipeIngredient.create!(recipe_id: @recipe_3.id, ingredient_id: @ingredient_1.id)
+      @rec_ing_13 = RecipeIngredient.create!(recipe_id: @recipe_3.id, ingredient_id: @ingredient_2.id)
+      @rec_ing_14 = RecipeIngredient.create!(recipe_id: @recipe_3.id, ingredient_id: @ingredient_7.id)
+      @rec_ing_15 = RecipeIngredient.create!(recipe_id: @recipe_3.id, ingredient_id: @ingredient_8.id)
+      @rec_ing_16 = RecipeIngredient.create!(recipe_id: @recipe_3.id, ingredient_id: @ingredient_9.id)
+    end
+
+    it "find all ingredients for a recipe" do
+      expected_1 = [@rec_ing_1, @rec_ing_2, @rec_ing_3, @rec_ing_4, @rec_ing_5, @rec_ing_6]
+      expected_2 = [@rec_ing_7, @rec_ing_8, @rec_ing_9, @rec_ing_10, @rec_ing_11]
+      expected_3 = [@rec_ing_12, @rec_ing_13, @rec_ing_14, @rec_ing_15, @rec_ing_16]
+
+      expect(RecipeIngredient.find_ingredients(@recipe_1.id)).to eq(expected_1)
+      expect(RecipeIngredient.find_ingredients(@recipe_2.id)).to eq(expected_2)
+      expect(RecipeIngredient.find_ingredients(@recipe_3.id)).to eq(expected_3)
```

Comment:
The danger of adding this model method to your model instead of relying on associations is that you have to do model tests like these, that are already solved for by _using_ those associations. If you find yourself writing tests like these, remember to double-check your associations and let Rails do the work for you!